src/vs/base/node/powershell.ts

```diff
@@ -8,18 +8,72 @@ import * as os from 'os';
 import * as path from 'vs/base/common/path';
 import { env } from 'vs/base/common/process';
 
-const WindowsPowerShell64BitLabel = 'Windows PowerShell';
-const WindowsPowerShell32BitLabel = 'Windows PowerShell (x86)';
-
 // This is required, since parseInt("7-preview") will return 7.
 const IntRegex: RegExp = /^\d+$/;
 
 const PwshMsixRegex: RegExp = /^Microsoft.PowerShell_.*/;
 const PwshPreviewMsixRegex: RegExp = /^Microsoft.PowerShellPreview_.*/;
 
-// The platform details descriptor for the platform we're on
-const isProcess64Bit: boolean = process.arch === 'x64';
-const isOS64Bit: boolean = isProcess64Bit || os.arch() === 'x64';
+enum Arch {
+	x64,
+	x86,
+	ARM
+}
+
+let processArch: Arch;
+switch (process.arch) {
+	case 'ia32':
+	case 'x32':
+		processArch = Arch.x86;
+		break;
+	case 'arm':
+	case 'arm64':
+		processArch = Arch.ARM;
+		break;
+	default:
+		processArch = Arch.x64;
+		break;
+}
+
+/*
+Currently, here are the values for these environment variables on their respective archs:
+
+On x86 process on x86:
+PROCESSOR_ARCHITECTURE is X86
+PROCESSOR_ARCHITEW6432 is undefined
+
+On x86 process on x64:
+PROCESSOR_ARCHITECTURE is X86
+PROCESSOR_ARCHITEW6432 is AMD64
+
+On x64 process on x64:
+PROCESSOR_ARCHITECTURE is AMD64
+PROCESSOR_ARCHITEW6432 is undefined
+
+On ARM process on ARM:
+PROCESSOR_ARCHITECTURE is ARM64
+PROCESSOR_ARCHITEW6432 is undefined
+
+On x86 process on ARM:
+PROCESSOR_ARCHITECTURE is X86
+PROCESSOR_ARCHITEW6432 is ARM64
+
+On x64 process on ARM:
+PROCESSOR_ARCHITECTURE is ARM64
+PROCESSOR_ARCHITEW6432 is undefined
+*/
+let osArch: Arch;
+if (process.env['PROCESSOR_ARCHITEW6432']) {
+	osArch = process.env['PROCESSOR_ARCHITEW6432'] === 'ARM64'
+		? Arch.ARM
+		: Arch.x64;
```

Comment:
Isn't this meant to be x86?

> On x86 process on ARM:
> PROCESSOR_ARCHITECTURE is X86
> PROCESSOR_ARCHITEW6432 is ARM64

Reply:
No. If `PROCESSOR_ARCHITEW6432` is ever defined it's because we are either on ARM64 or x64 because that environment variable is only defined when a 32bit process is running on a different architecture than 32bit.